docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -0,0 +1,163 @@
+        false); // transpose
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+*Before the preceding function has been run.*
+
+![Data in Excel before range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-before.png)
+
+*After the preceding function has been run.*
+
+![Data in Excel after range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-after.png)
+
+`transpose` determines whether or not the data is transposed, meaning its rows and columns are switched, into the source location.
+A transposed range is flipped along the main diagonal, so rows **1**, **2**, and **3** will become columns **A**, **B**, and **C**.
+
+## Remove duplicates
+
+> [!NOTE]
+> The `Range.removeDuplicates` function is currently available only in public preview (beta). To use this feature, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+> If you are using TypeScript or your code editor uses TypeScript type definition files for IntelliSense, use https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts.
+
+`Range.removeDuplicates` removes rows with duplicate entries in the specified columns. The function goes through each row in the range from top (the lowest index row in the range) to bottom (the highest valued index in the range). If a value in the specified column or columns is encountered more than once, that value's row is deleted. Subsequent rows in the range are shifted up. `removeDuplicates` does not affect the position of cells outside of the range.
+
+`Range.removeDuplicates` takes in a `number[]` representing the column indices which are checked for duplicates. This array is zero-based and relative to the range, not the worksheet. The function also takes in a boolean argument, `includesHeader` to determine if the range's top row is a header. When **true**, the top row is ignored when considering duplicates. A `RemoveDuplicatesResult` is returned upon completion of `removeDuplicates`, containing the number of rows removed and the number of unique rows remaining.
+
+`Range.removeDuplicates` considers cell values, not function results. If two different functions evaluate to the same result, they are not considered duplicates. Relatedly, blank cells are checked by `removeDuplicates`. Any blank rows will be included in the `RemoveDuplicatesResult`.
+
+The following sample shows the removal of entries with duplicate values in the first column.
+
+```js
+Excel.run(async (context) => {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    var range = sheet.getRange("B2:D11");
+
+    var deleteResult = range.removeDuplicates([0],true);
+    deleteResult.load();
+
+  return context.sync()
+    .then(function () {
+        console.log(deleteResult.removed + " entries with duplicate names removed.");
+        console.log(deleteResult.uniqueRemaining + " entries with unique names remain in the range.");
+});
+```
```

Comment:
Indentation is inconsistent in this code sample (should use either 2 spaces or 4 spaces consistently throughout).